apps/src/netsim/NetSimRouterNode.js

```diff
@@ -299,6 +315,15 @@ NetSimRouterNode.prototype.getDisplayName = function () {
   });
 };
 
+/**
+ * Get node's hostname, a modified version of its display name.
+ * @returns {string}
+ * @override
+ */
+NetSimRouterNode.prototype.getHostname = function () {
+  return this.getDisplayName().replace(/[^\w\d]/g, '').toLowerCase();
```

Comment:
I like explanations of regexes. In this case, it seems like an explanation of why were removing these leading characters also seems worthwhile.